plugins/jobtype/src/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -0,0 +1,409 @@
+/*
+ * Copyright 2012 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.jobtype;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.hadoop.yarn.api.records.ApplicationId;
+import org.apache.hadoop.yarn.client.api.YarnClient;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+import org.apache.log4j.Logger;
+
+import azkaban.security.commons.HadoopSecurityManager;
+import azkaban.security.commons.HadoopSecurityManagerException;
+import azkaban.utils.Props;
+
+/**
+ * There are many common methods that's required by the Hadoop*Job.java's. They are all consolidated
+ * here.
+ * 
+ * @see azkaban.jobtype.HadoopSparkJob
+ * @see azkaban.jobtype.HadoopHiveJob
+ * @see azkaban.jobtype.HadoopPigJob
+ * @see azkaban.jobtype.HadoopJavaJob
+ */
+
+public class HadoopJobUtils {
+
+  public static final String HADOOP_SECURITY_MANAGER_CLASS_PARAM = "hadoop.security.manager.class";
+
+  public static void cancelHadoopTokens(HadoopSecurityManager hadoopSecurityManager,
+          String userToProxy, File tokenFile, Logger log) {
+    try {
+      hadoopSecurityManager.cancelTokens(tokenFile, userToProxy, log);
+    } catch (HadoopSecurityManagerException e) {
+      e.printStackTrace();
+      log.error(e.getCause() + e.getMessage());
+    } catch (Exception e) {
+      e.printStackTrace();
+      log.error(e.getCause() + e.getMessage());
+    }
+  }
+
+  /**
+   * Based on the HADOOP_SECURITY_MANAGER_CLASS_PARAM setting in the incoming props, finds the
+   * correct HadoopSecurityManager Java class
+   * 
+   * @param props
+   * @param log
+   * @return a HadoopSecurityManager object. Will throw exception if any errors occur (including not
+   *         finding a class)
+   * @throws RuntimeException
+   *           : If any errors happen along the way.
+   */
+  public static HadoopSecurityManager loadHadoopSecurityManager(Props props, Logger log)
```

Comment:
would it sometimes be helpful to have the log so that we can print log.info out from this method? (in this case)